pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -106,9 +106,7 @@ func (l *LeaderElection) create() (*leaderelection.LeaderElector, error) {
 		// When Pilot exits, the lease will be dropped. This is more likely to lead to a case where
 		// to instances are both considered the leaders. As such, if this is intended to be use for mission-critical
 		// usages (rather than avoiding duplication of work), this may need to be re-evaluated.
-		// TODO (therealmitchconnors) move background analysis to leader instance once this bug is fixed
-		// TODO this should be true once https://github.com/kubernetes/kubernetes/issues/87800 is fixed
-		ReleaseOnCancel: false,
+		ReleaseOnCancel: true,
```

Comment:
```
	// ReleaseOnCancel should be set true if the lock should be released
	// when the run context is cancelled. If you set this to true, you must
	// ensure all code guarded by this lease has successfully completed
	// prior to cancelling the context, or you may have two processes
	// simultaneously acting on the critical path.
```

As the comments says, all run fuctions should be well stopped, but i can find at least validation server does not.

Reply:
this is only triggered when pilot shuts down so it shouldn't be an issue. And the leader election is just an optimization, if multiple processes run it there is no negative impact

Reply:
This is the logic now:

istio.LeaderElection.Run(dead loop) -> k8s,LeaderElector.Run() -》 acquire lock --> OnStartedLeading->renew lock  

When renew failed, call release ---》 OnStopppedLeading 




If it lose the lock, and the registered functions should exit, otherwise when this istiod acquires lock again, there will be two or more  go routines running the registerd functions.

We need to do some fix on the application level logic.

Reply:
I am confused how this is changing like you are saying. My understanding is this will just drop the lock when istiod exits, allowing another pod to pick it up immediately instead of 30s later

Reply:
> allowing another pod to pick it up immediately
 `another pod ` can be itself too, whatever there maybe multiple istiod working at the same time working patching validationWebhook config.

Though this is not related much to this pr. and I have found another small bug. Will fix later.